traits/trait_types.py

```diff
@@ -1430,6 +1431,21 @@ def validate(self, object, name, value):
 
             Note: The 'fast validator' version performs this check in C.
         """
+        try:
+            # Note: Python 3.5 does not implement __fspath__
+            from os import fspath
```

Comment:
I think it would be better to move the `try` / `except` for the `ImportError` to the top of the module, and define a "_HAVE_OS_FSPATH" constant (a better name almost certainly exists). Doing a failed import on every validation is unnecessarily expensive.